test/conf.spec.js

```diff
@@ -54,6 +59,57 @@ describe('JSDoc configuration options', () => {
             .finally(done);
     });
 
+    it('config.tags.allowUnknownTags should override options.allowUnknownTags and throw', done => {
```

Comment:
Added 3 tests, the latter simply ensuring that custom options can be passed without side effects. An enhancement to the third test could include the ability to set `keep: true` on the config file, read it into a buffer and verify it's contents are what's expected, but I felt that update didn't fit with this PR and issue.